stepactions/eaas-copy-secrets-to-ephemeral-cluster/0.1/eaas-copy-secrets-to-ephemeral-cluster.yaml

```
@@ -0,0 +1,42 @@
+apiVersion: tekton.dev/v1alpha1
+kind: StepAction
+metadata:
+  name: eaas-copy-secrets-to-ephemeral-cluster
+spec:
+  description: >-
+    This StepAction copies Secrets from the current namespace into a configurable namespace on an
+    ephemeral cluster. The name and content of each Secret is unaltered in the process.
+  image: quay.io/redhat-appstudio/appstudio-utils@sha256:586149e3f18d966f681d956ab074b4e1d8433663d615ed86e19a3804ba952dfe
+  params:
+    - name: credentials
+      type: string
+      description: A volume containing credentials to the remote cluster
+    - name: kubeconfig
+      type: string
+      description: Relative path to the kubeconfig in the mounted cluster credentials volume
+    - name: namespace
+      type: string
+      description: The destination namespace for the secrets. The namespace must already exist.
+    - name: labelSelector
+      type: string
+      description: A label selector identifying the secrets to be copied
+  env:
+    - name: DEST_KUBECONFIG
+      value: "/credentials/$(params.kubeconfig)"
+    - name: NAMESPACE
+      value: "$(params.namespace)"
+    - name: LABEL_SELECTOR
+      value: "$(params.labelSelector)"
+  volumeMounts:
+    - name: "$(params.credentials)"
+      mountPath: /credentials
+  script: |
+    #!/bin/bash
+    set -eo pipefail
+
+    oc get secrets -l "$LABEL_SELECTOR" -o json \
+      | jq ".items[].metadata.namespace=\"$NAMESPACE\"" \
```

jq tip: you can use `--arg argname "$value"` to pass variables to jq in a safer way (this won't do unexpected things if `$NAMESPACE` contains double-quotes, for example)
```suggestion
      | jq --arg ns "$NAMESPACE" '.items[].metadata.namespace=$ns' \
```

---

Nice tip! Updated.